### PR TITLE
unify configuration (use environment variables) and dockerize

### PR DESCRIPTION
**Pull request by cirroskais**
Sponsored by MadHouse Labs:tm:

### CODE REVIEW

**Comment by nbitzz** [comment]
added #22 here because this solves it by doing the exact opposite of what #22 suggests

**Comment by cirroskais** [comment]
almost ready for review; only issue currently standing is that assets are not copied over to the container
image

File: src/server/routes/api/apis.ts
Comment (cirroskais) [inline src/server/routes/api/apis.ts:1]: image

**Comment by nbitzz** [comment]
Man i should just merge this shoulndt i

**Comment by nbitzz** [comment]
@Jack5079 Oi

**Comment by Jack5079** [comment]
So damn true!

**Comment by cirroskais** [comment]
Lgtm

**Comment by nbitzz** [comment]
i dotn fgjuicking know just merge it

**Comment by nbitzz** [comment]
We can like fix the API mounting later right now idgaf

**Comment by cirroskais** [comment]
error while creating account: 
```
file:///usr/src/app/node_modules/@hono/node-server/dist/index.mjs:151
  for (const [k, v] of headers) {
                       ^

TypeError: headers is not iterable
    at buildOutgoingHttpHeaders (file:///usr/src/app/node_modules/@hono/node-server/dist/index.mjs:151:24)
    at responseViaResponseObject (file:///usr/src/app/node_modules/@hono/node-server/dist/index.mjs:315:27)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
```

**Comment by cirroskais** [comment]
currently deployed at https://monofile.dev.madhouselabs.net/ go crazy

**Comment by cirroskais** [comment]
```
TypeError: AuthTokens.find is not a function
    at getToken (file:///usr/src/app/out/server/lib/auth.js:36:23)
    at Module.validate (file:///usr/src/app/out/server/lib/auth.js:39:12)
    at Module.getFromToken (file:///usr/src/app/out/server/lib/accounts.js:49:22)
    at getAccount (file:///usr/src/app/out/server/lib/middleware.js:9:33)
    at dispatch (file:///usr/src/app/node_modules/hono/dist/compose.js:29:23)
    at file:///usr/src/app/node_modules/hono/dist/compose.js:6:12
    at file:///usr/src/app/node_modules/hono/dist/hono-base.js:226:31
    at Hono.dispatch (file:///usr/src/app/node_modules/hono/dist/hono-base.js:236:7)
    at fetch (file:///usr/src/app/node_modules/hono/dist/hono-base.js:60:19)
    at Server.<anonymous> (file:///usr/src/app/node_modules/@hono/node-server/dist/index.mjs:362:13)
```

**Comment by nbitzz** [comment]
AAAAAAAAAAAAAAAAAAAAa

**Comment by nbitzz** [comment]
> error while creating account:
> 
> ```
> file:///usr/src/app/node_modules/@hono/node-server/dist/index.mjs:151
>   for (const [k, v] of headers) {
>                        ^
> 
> TypeError: headers is not iterable
>     at buildOutgoingHttpHeaders (file:///usr/src/app/node_modules/@hono/node-server/dist/index.mjs:151:24)
>     at responseViaResponseObject (file:///usr/src/app/node_modules/@hono/node-server/dist/index.mjs:315:27)
>     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
> ```

Pull latest commit to api-v1

**Comment by cirroskais** [comment]
deployed

**Comment by nbitzz** [comment]
> ```
> TypeError: AuthTokens.find is not a function
>     at getToken (file:///usr/src/app/out/server/lib/auth.js:36:23)
>     at Module.validate (file:///usr/src/app/out/server/lib/auth.js:39:12)
>     at Module.getFromToken (file:///usr/src/app/out/server/lib/accounts.js:49:22)
>     at getAccount (file:///usr/src/app/out/server/lib/middleware.js:9:33)
>     at dispatch (file:///usr/src/app/node_modules/hono/dist/compose.js:29:23)
>     at file:///usr/src/app/node_modules/hono/dist/compose.js:6:12
>     at file:///usr/src/app/node_modules/hono/dist/hono-base.js:226:31
>     at Hono.dispatch (file:///usr/src/app/node_modules/hono/dist/hono-base.js:236:7)
>     at fetch (file:///usr/src/app/node_modules/hono/dist/hono-base.js:60:19)
>     at Server.<anonymous> (file:///usr/src/app/node_modules/@hono/node-server/dist/index.mjs:362:13)
> ```

Man i don't even know 😭

**Comment by nbitzz** [comment]
> > > ```
> > > TypeError: AuthTokens.find is not a function
> > >     at getToken (file:///usr/src/app/out/server/lib/auth.js:36:23)
> > >     at Module.validate (file:///usr/src/app/out/server/lib/auth.js:39:12)
> > >     at Module.getFromToken (file:///usr/src/app/out/server/lib/accounts.js:49:22)
> > >     at getAccount (file:///usr/src/app/out/server/lib/middleware.js:9:33)
> > >     at dispatch (file:///usr/src/app/node_modules/hono/dist/compose.js:29:23)
> > >     at file:///usr/src/app/node_modules/hono/dist/compose.js:6:12
> > >     at file:///usr/src/app/node_modules/hono/dist/hono-base.js:226:31
> > >     at Hono.dispatch (file:///usr/src/app/node_modules/hono/dist/hono-base.js:236:7)
> > >     at fetch (file:///usr/src/app/node_modules/hono/dist/hono-base.js:60:19)
> > >     at Server.<anonymous> (file:///usr/src/app/node_modules/@hono/node-server/dist/index.mjs:362:13)
> > > ```
> > 
> > 
> > Man i don't even know 😭
> 
> i was stupid and set tokens.json to " {} " and not " [] " my fault og
> 
> you should probably make it... make those files. if they dont exist

if it fails it just. doesn't gaf and sets the authtoken array to [] so like it'll happen eventually anywaaaay

**Comment by nbitzz** [comment]
We should really have just used SQL why didnt we monofile from scratch

**Comment by nbitzz** [comment]
Ok so yeah it basically works we can merge right

**Comment by nbitzz** [comment]
Switching to mogno

**Comment by cirroskais** [comment]
LGTM

**Comment by cirroskais** [comment]
@nbitzz mergeeeee

**Comment by nbitzz** [comment]
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHH

**Comment by nbitzz** [comment]
@Jack5079 Monobot for automatic deployments of monofile whwene>>
Maybe i should do that actually

**Pull request merged**